Add tests for `script.evaluate` for serialization of primitive/simple values.

lutien opened the pull request:
Add more tests for `script.evaluate` for serialization of primitive/simple values.

lutien committed:
Format tests for `script.evaluate`

lutien committed:
Add tests for `script.evaluate` for serialization of primitive/simple values.

juliandescottes commented on webdriver/tests/bidi/script/evaluate/evaluate.py:
```diff
@@ -6,35 +6,49 @@
+)
+async def test_eval(bidi_session, top_context, expression, result):
+    assert result == await bidi_session.script.evaluate(
+        expression=expression, target=ContextTarget(top_context["context"])
```
await_promise is mandatory now
```suggestion
        expression=expression, target=ContextTarget(top_context["context"]), await_promise=False
```

lutien replied:
Oh, yes, haven't noticed it, returned but with `True` value as it used to be.

juliandescottes commented on webdriver/tests/bidi/script/evaluate/evaluate.py:
```diff
@@ -6,35 +6,49 @@
+    ],
+)
+async def test_eval(bidi_session, top_context, expression, result):
+    assert result == await bidi_session.script.evaluate(
```
nitpick: Maybe keep the same naming as the other tests, and have an `expected` parameter instead of `result` Then `result = await evaluate()` and `assert result == expected`

juliandescottes reviewed (changes requested):
Thanks! One issue to fix + nit otherwise looks good to me

lutien committed:
Return `await_promise` and improve on parameter naming

sadym-chromium commented on webdriver/tests/bidi/script/evaluate/evaluate.py:
```diff
@@ -6,35 +6,53 @@
+    [
+        ("undefined", {"type": "undefined"}),
+        ("null", {"type": "null"}),
+        ("'foo'+ 'bar'", {"type": "string", "value": "foobar"}),
```
What about `String` object?

lutien replied:
Do you mean something like `new String("foo")`? Then it'll be type `object` 🤔

sadym-chromium commented on webdriver/tests/bidi/script/evaluate/evaluate.py:
```diff
@@ -6,35 +6,53 @@
+        ("-0", {"type": "number", "value": "-0"}),
+        ("Infinity", {"type": "number", "value": "+Infinity"}),
+        ("-Infinity", {"type": "number", "value": "-Infinity"}),
+        ("1 + 2", {"type": "number", "value": 3}),
```
float?

lutien replied:
added ✅

sadym-chromium commented on webdriver/tests/bidi/script/evaluate/evaluate.py:
```diff
@@ -6,35 +6,53 @@
+    [
+        ("undefined", {"type": "undefined"}),
+        ("null", {"type": "null"}),
+        ("'foo'+ 'bar'", {"type": "string", "value": "foobar"}),
```
I'd keep is simple. Like `'foobar'` instead of `'foo' + 'bar'`.

lutien replied:
fixed ✅

sadym-chromium reviewed (commented):
Having need in testing non-primitives, and deserialization, consider moving the test to a separate file, or even folder like `webdriver/tests/bidi/serialization/primitive.py`.

lutien committed:
Move tests for serialization to a separate file

lutien committed:
Add float test case and fix string test case

lutien commented:
@sadym-chromium, I've moved this test to a separate file

lutien committed:
Revert "Format tests for `script.evaluate`"

This reverts commit e4c5cab2109ef60e59730a0a8d1bb90ceb60fb89.

lutien commented:
> > @sadym-chromium, I've moved this test to a separate file
> 
> Maybe revert reformatting `webdriver/tests/bidi/script/evaluate/evaluate.py` as not relevant to the goal of the PR?

Sure, reverted

sadym-chromium commented on webdriver/tests/bidi/script/evaluate/serialization.py:
```diff
@@ -0,0 +1,30 @@
+        ("-0", {"type": "number", "value": "-0"}),
+        ("Infinity", {"type": "number", "value": "+Infinity"}),
+        ("-Infinity", {"type": "number", "value": "-Infinity"}),
+        ("1 + 2", {"type": "number", "value": 3}),
+        ("1 + 0.4", {"type": "number", "value": 1.4}),
+        ("1 === 1", {"type": "boolean", "value": True}),
+        ("1 !== 1", {"type": "boolean", "value": False}),
```
Sorry for not being clear enough. I meant simplification in other lines as well. I guess it's way more readable. Not a blocker though.
```suggestion
        ("3", {"type": "number", "value": 3}),
        ("1.4", {"type": "number", "value": 1.4}),
        ("true", {"type": "boolean", "value": True}),
        ("false", {"type": "boolean", "value": False}),
```

lutien replied:
Yeah, you're right. Especially since it's only about serialization now, it can be simpler. 
I've updated it.

lutien committed:
Simplify expressions

juliandescottes reviewed (approved):
Great, thanks for all the updates @lutien

lutien committed:
Use Number.NaN instead of expression
